packages/openapi3-parser/lib/parser/oas/parseSchemaObject.js

```diff
@@ -404,8 +412,9 @@ function parseSchema(context) {
         element.description = description;
       }
 
+      // On OAS 3.0, nullable is a keyword, on OAS 3.1, null goes in type
       const nullable = schema.getValue('nullable');
-      if (nullable) {
+      if (nullable || (type.includes('null') && element.element !== 'null')) {
```

Comment:
We want to add nullable, only if the type isn't null.

`type: null` causes null element, whereas `type: [string, null]` means string element, that this branch makes nullable.